File: T6_herencia/src/Trabajadores/Autonomo.java
package Trabajadores;

import java.util.ArrayList;

public final class Autonomo extends Trabajador{
    private double cuotas;
    public Autonomo(String nombre, String apellido, String dni, double sueldo, boolean contratado, double cuotas) {
        super(nombre, apellido, dni, sueldo, contratado);
        this.cuotas = cuotas;
    }

    @Override
    public void MostrarDatos() {
        super.MostrarDatos();
        System.out.println("Sueldo: "+getSueldo());
        System.out.println("Contratado: "+isContratado());
        System.out.println("Numero de pagas: "+cuotas);
    }

    public Autonomo() {
    }

    public double getCuotas() {
        return cuotas;
    }

    public void setCuotas(double cuotas) {
        this.cuotas = cuotas;
    }
}

File: DAM2/T1_INICIO/Trabajadores/src/main/kotlin/model/Autonomo.kt
package model

class Autonomo (nombre: String, apellido: String, dni: String, sueldo: Int=10000, var contratado:Boolean=false,
                var cuotaSS:Int):
    Trabajador(nombre,apellido,dni, sueldo){
    override fun calcularSueldoNetoMes() {
    var salario = (sueldo-(cuotaSS*12))/12
    }
    override fun mostrarDatos() {
        super.mostrarDatos()
    println("Cuota Seguridad Social $cuotaSS")
        println("Contratado $contratado")
    }

}